lib/portage/_sets/files.py

```diff
@@ -176,6 +177,11 @@ def multiBuilder(self, options, settings, trees):
             directory = normalize_path(directory)
 
             for parent, dirs, files in os.walk(directory):
+                if portage.utf8_mode:
+                    parent = os.fsencode(parent)
+                    dirs = [os.fsencode(value) for value in dirs]
```

Comment:
There's a `dirs.remove(d)` that will be ineffective because this is not the dirs used by os.walk.

Reply:
Added an `omit_dir` lambda to handle it.